setuptools/tests/test_config_discovery.py

```diff
@@ -255,7 +255,7 @@ def test_py_modules_when_wheel_dir_is_cwd(self, tmp_path):
 
 
 class TestNoConfig:
-    DEFAULT_VERSION = "0.0.0"  # Default version given by setuptools
+    CANONICAL_DEFAULT_VERSION = "0"  # Canonical default version given by setuptools
```

Comment:
TestNoConfig doesn't feel like the right place to assert expectations about file naming. I suspect this test probably needs to be adjusted to account for the change, but it also feels like there should be another test asserting specifically the intention about how sdists are named. It's possible there are tests in distutils that cover this behavior and also need to be adjusted (but aren't run as part of setuptools).